src/main/ApplicationImpl.cpp

```diff
@@ -236,6 +312,7 @@ ApplicationImpl::newDB()
 {
     mDatabase->initialize();
     mDatabase->upgradeToCurrentSchema();
+    maybeRebuildLedger(*this, false);
```

Comment:
grep for `upgradeToCurrentSchema`: there are two other call sites.
I *think* we should factor this into its own method so that we never forget to call `maybeRebuildLedger` after upgrading the schema.

Reply:
My implementation used to look like this, but something went wrong in the rebase (lot of changes to this code lately). Good catch, fixed by:
- creating a method that bundles schema upgrade and ledger rebuild
- eliminating the third call site, rerouting that to the other code paths